### PR TITLE
Turn one of the M9K shotguns into a low-damage, high-knockback gun

### PR DESCRIPTION
**Pull request by legokidlogan**

### CODE REVIEW

File: lua/cfc_entity_stubber/cfc_m9k/cfc9k_shotgun_bonk.lua [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+        if not IsValid( wep ) then return end
+        if wep:GetClass() ~= BONK_GUN_CLASS then return end
+
+        bonkVictim( attacker, victim, dmg, wep )
+    end )
+
+
+    if not IMPACT_DAMAGE_ENABLED then return end
+
+    hook.Add( "Think", "M9K_Stubber_BonkGun_DetectImpact", function()
```
Comment (wrefgtzweve) [inline lua/cfc_entity_stubber/cfc_m9k/cfc9k_shotgun_bonk.lua:294]: shouldn't be on client

File: lua/cfc_entity_stubber/cfc_m9k/cfc9k_shotgun_bonk.lua [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        end
+
+        if AIR_SHOTS_REFUND_AMMO then
+            ply.cfc9k_bonkCanRefund = true
+        end
+
+        return self:_ShootBullet( damage, recoil, numBullets, spread )
+    end
+
+    hook.Add( "EntityTakeDamage", "M9K_Stubber_BonkGun_YeetVictim", function( victim, dmg )
```
Comment (wrefgtzweve) [inline lua/cfc_entity_stubber/cfc_m9k/cfc9k_shotgun_bonk.lua:274]: shouldn't be on client

File: lua/cfc_entity_stubber/cfc_m9k/cfc9k_bonk.lua [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+        end
+
+        return
+    end
+
+    handleImpact( ent, accel )
+end
+
+
+hook.Add( "EntityTakeDamage", "M9K_Stubber_Bonk_YeetVictim", function( victim, dmg )
```
Comment (brandonsturgeon) [inline lua/cfc_entity_stubber/cfc_m9k/cfc9k_bonk.lua:321]: Probably better in PostEntityTakeDamage?

File: lua/cfc_entity_stubber/cfc_m9k/cfc9k_bonk.lua [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+        if not IsValid( attacker ) then return end
+        attacker.cfc9k_bonkCannotRefund = false
+    end )
+end
+
+local function counteractOpposingVelocity( ply, forceDir )
+    local plyVel = ply:GetVelocity()
+    local dot = plyVel:Dot( forceDir )
+
+    if dot >= 0 then return Vector( 0, 0, 0 ) end
```
Comment (brandonsturgeon) [inline lua/cfc_entity_stubber/cfc_m9k/cfc9k_bonk.lua:125]: Localize this vector at the top if it won't be modified

File: lua/cfc_entity_stubber/cfc_m9k/cfc9k_bonk.lua [hunk 5]
```diff
@@ -0,0 +1,344 @@
+AddCSLuaFile()
+
```
Comment (brandonsturgeon) [inline lua/cfc_entity_stubber/cfc_m9k/cfc9k_bonk.lua:2]: Localize IsValid and prefer `:IsValid()`

**Review by brandonsturgeon — CHANGES_REQUESTED** [review]

**Review by brandonsturgeon — APPROVED** [review]

**Pull request merged**